homeassistant/components/kulersky/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for Kuler Sky."""
+import logging
+
+import pykulersky
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ADDRESS, CONF_NAME
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Kuler Sky"
+
+
+class KulerSkyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Kuler sky config flow."""
+
+    async def async_step_user(self, user_input=None):
+        """Search for nearby lights and prompt user."""
+        errors = {}
+
+        if user_input is not None:
+            address, name = user_input["device"].split(" ", 1)
+
+            await self.async_set_unique_id(address)
+            self._abort_if_unique_id_configured()
+
+            # Ensure the light can connect
+            light = pykulersky.Light(address, name)
+            try:
+                await self.hass.async_add_executor_job(light.connect)
+                return self.async_create_entry(
+                    title=name,
+                    data={
+                        CONF_ADDRESS: address,
+                        CONF_NAME: name,
+                    },
+                )
+            except pykulersky.PykulerskyException:
+                errors["device"] = "cannot_connect"
+
+        try:
+            devices = await self.hass.async_add_executor_job(pykulersky.discover)
+        except pykulersky.PykulerskyException as exc:
+            _LOGGER.error("Exception scanning for Kuler Sky devices", exc_info=exc)
+            return self.async_abort(reason="scan_error")
+
+        options = [
```

Comment:
Why allow the user to pick? We could replace this config flow with https://developers.home-assistant.io/docs/config_entries_config_flow_handler#discoverable-integrations-that-require-no-authentication and then have the light platform set them all up. The user can disable the entities of the lights they don't want.

Reply:
So I could have sworn I commented this somewhere, but I don't see it. :woman_facepalming: The problem with these devices is they don't expose any identifying information, at least nothing available to the underlying `pygatt` library. So the best that `pykulersky` can do is discover all nearby bluetooth devices. (I need to update that library `discover` method name to make it more clear) So the options we've got are either have the user pick the correct device, or attempt to connect to each discovered bluetooth device, which could potentially leave the user waiting at the spinner for a while if there are a lot of other bluetooth devices in the vicinity. I'm OK with either option, but those are the tradeoffs.

Reply:
Thanks for the review!

I pushed up some comments, and renamed the upstream method so that the current behavior is at least more clear when reading the code. Still happy to switch to plan B if you'd prefer.

Reply:
The discovery config flow only requires a single light to be discovered to set up the config flow. (Just wants to make sure a device can be found).

After that it finishes the flow and lights will be added to Home Assistant as discovery picks up more.

Reply:
I think that having the user pick a bluetooth address is putting burden on the user. Having a user wait while we figure out if they have any device is a lot better experience compared to having the user set up each individual device and having to figure out their addresses too.

Reply:
Even discovering the first light could be really slow though, since we'll have to attempt a connection with every nearby bluetooth device before we know if it's a valid device. I'll switch it over though. Sounds like simple but slow is the preferred solution.

Reply:
Maybe we can change the discovery flow to just check if Bluetooth works ?

Reply:
Oh, I like that idea! Should be a quick test to ensure the bluetooth stack is working, and the devices will be available once they're discovered. :+1: